oauth/src/test/java/org/mule/test/module/extension/oauth/clientcredentials/DynamicOAuthClientCredentialsExtensionTestCase.java

```diff
@@ -72,8 +82,39 @@ public void getSecondConnectionwithDynamicConfig() throws Exception {
     verifyCall(ANOTHER_CLIENT_ID, ANOTHER_CLIENT_SECRET, ANOTHER_SCOPES);
   }
 
-  private void getConnectionFlow(String clientId, String clientSecret, String scopes) throws Exception {
-    flowRunner("getConnection")
+  @Test
+  @Description("Create two dynamic configurations with different values and check their access tokens do not overwrite each other.")
+  public void dynamicConfigsUseDifferentTokens() throws Exception {
+    CoreEvent firstEvent = getConnectionFlow(CLIENT_ID, CLIENT_SECRET, SCOPES);
+    TestOAuthConnectionState firstConnectionState =
+        ((TestOAuthConnection) firstEvent.getMessage().getPayload().getValue()).getState();
+    ClientCredentialsConfig config = (ClientCredentialsConfig) muleContext.getExtensionManager()
+        .getConfiguration("oauth", firstEvent).getValue();
+
+    WireMock.reset();
+    wireMock.stubFor(post(urlPathMatching("/" + TOKEN_PATH))
+        .willReturn(aResponse()
+            .withStatus(OK.getStatusCode())
+            .withBody(accessTokenContent(REFRESH_TOKEN))
+            .withHeader(CONTENT_TYPE, "application/json")));
+    TestOAuthConnectionState secondConnectionState =
+        ((TestOAuthConnection) getConnectionFlow(ANOTHER_CLIENT_ID, ANOTHER_CLIENT_SECRET, ANOTHER_SCOPES).getMessage()
+            .getPayload().getValue()).getState();
+
+    new PollingProber(10000, 1000).check(new JUnitLambdaProbe(() -> {
+      assertThat(config.getDispose(), is(1));
+      return true;
+    }, "config was not disposed")); // Wait for the dynamic config to expire so that another connection is created, but the underlying dancer is still the same.
+    TestOAuthConnectionState thirdConnectionState =
+        ((TestOAuthConnection) getConnectionFlow(CLIENT_ID, CLIENT_SECRET, SCOPES).getMessage().getPayload().getValue())
+            .getState();
+
+    assertThat(firstConnectionState.getState().getAccessToken(), is(thirdConnectionState.getState().getAccessToken()));
```

Comment:
isn't it better and more clear to assert the expected token on each state?

Reply:
Possibly, yeah.